core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SnapshotApiServiceSpec.scala

```diff
@@ -155,41 +155,51 @@ class SnapshotApiServiceSpec extends ApiServiceSpec {
       check { assertResult(StatusCodes.NotFound) {status} }
   }
 
-  it should "return 200 when a user lists all workspaces" in withTestDataApiServicesAndUser(testData.userReader.userEmail.value) { services =>
-    Get(s"${testData.wsName.path}/snapshots", httpJson(
-      EnumerateSnapshotRequestBody(
-        offset = 0,
-        limit = 10
+  it should "return 200 when a user lists all snapshots in a workspace" in withTestDataApiServices { services =>
+    // First, create two workspaces
+    Post(s"${testData.wsName.path}/snapshots", httpJson(
+      DataRepoSnapshot(
+        name = "foo",
+        snapshotId = "realsnapshot"
       )
     )) ~>
       sealRoute(services.snapshotRoutes) ~>
       check {
-        val response = responseAs[DataRepoSnapshotList]
-        assertResult(StatusCodes.OK) {status}
-        assert(response.snapshots.size == 0)
+        val response = responseAs[DataRepoSnapshotReference]
+        assertResult(StatusCodes.Created) {status}
+        Post(s"${testData.wsName.path}/snapshots", httpJson(
+          DataRepoSnapshot(
+            name = "bar",
+            snapshotId = "realsnapshot2"
+          )
+        )) ~>
+          sealRoute(services.snapshotRoutes) ~>
+          check {
+            val response = responseAs[DataRepoSnapshotReference]
+            assertResult(StatusCodes.Created) {status}
+            // Then, list them both
+            Get(s"${testData.wsName.path}/snapshots?offset=0&limit=10") ~>
+              sealRoute(services.snapshotRoutes) ~>
+              check {
+                val response = responseAs[DataRepoSnapshotList]
+                assertResult(StatusCodes.OK) {status}
+                assert(response.snapshots.size == 2)
+              }
+          }
       }
+
   }
```

Comment:
couple comments on this test:
1. My original thought was that you'd edit `MockWorkspaceManagerDAO` and hardcode some fixture data inside the mock, then all unit tests for read-only functions could request data from the mock, know what to expect, and assert on it. That way, a read-only unit test wouldn't fail if there's a flaw in functions that write. I'm fine either way and wouldn't hold the PR on this design.
2. In looking at `MockWorkspaceManagerDAO`, there's a problem with its `references` member: it applies to all workspaces. If you use `MockWorkspaceManagerDAO.createDataReference` to add a snapshot for workspace A, then use `MockWorkspaceManagerDAO.enumerateDataReferences` to list snapshots for workspace B, you'll get workspace A's snapshot. I feel like this is going to bite us and we should solve it soon; if you want to solve it in a follow-on PR, that's fine

Reply:
For 1, I'd lean towards using the write methods instead of hardcoding data - the unit tests are a bit less isolated, but there's one less place to maintain write behavior.
Agree on 2, I think it's enough of an issue to include in this PR. I updated the references map to use a `(workspaceId, referenceId)` tuple as a key instead of just `referenceId` which seems like a decent representation of the real system, but let me know if you'd rather take a different approach.